Update best practices for multi-cluster allocation (#4157)
googleforgames/agones: site/content/en/docs/Guides/Best Practices/_index.md

### Allocate Across Clusters

> Agones supports Multi-cluster Allocation to avoid a single point of failure when allocating game servers. While earlier versions of Agones included a custom multi-cluster allocation solution, the current best practice is to use a **Service Mesh** (e.g., Istio, Linkerd) to handle allocation traffic between clusters.

Reviewer: ```suggestion
Agones supports Multi-cluster Allocation to avoid a single point of failure when allocating game servers. While earlier versions of Agones included a custom multi-cluster allocation solution, the current best practice is to use a **Service Mesh** (e.g., Istio, Linkerd, [Google Cloud Service Mesh](https://cloud.google.com/service-mesh/docs/overview)) to handle allocation traffic between clusters.
```

> By deploying a Service Mesh in each of your Agones clusters and your backend services cluster, you can expose and route traffic to each cluster’s agones-allocator endpoint based on cluster priority, latency, or other criteria.

Reviewer: ```suggestion
By deploying a Service Mesh across your Agones clusters and backend services, you can expose and route traffic to each cluster’s agones-allocator endpoint based on cluster priority, latency, or other criteria.
```

> To implement this approach, refer to the full setup and guidance in the [Multi-cluster Allocation documentation](https://agones.dev/site/docs/advanced/multi-cluster-allocation/).

Reviewer: Nit: update to use the relref 

```suggestion
To implement this approach, refer to the full setup and guidance in the [Multi-cluster Allocation documentation]({{< relref "multi-cluster-allocation" >}}).
```

> You can also explore the [Global Multiplayer Demo](https://github.com/googleforgames/global-multiplayer-demo) for a working example using htGoogle Cloud Service Mesh with Istio.

Reviewer: ```suggestion
You can also explore the [Global Multiplayer Demo](https://github.com/googleforgames/global-multiplayer-demo) for a working example using Google Cloud Service Mesh with Istio.
```
